src/libs/sfdk/dockermanager.cpp

```diff
@@ -0,0 +1,129 @@
+/****************************************************************************
+**
+** Copyright (C) 2019 Open Mobile Platform LLC.
+** Contact: http://jolla.com/
+**
+** This file is part of Qt Creator.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU Lesser General Public License version 2.1 requirements
+** will be met: http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** Other Usage
+**
+** Alternatively, this file may be used in accordance with the terms and
+** conditions contained in a signed written agreement between you and
+** The Qt Company.
+**
+****************************************************************************/
+#include "dockermanager.h"
+
+#include "asynchronous_p.h"
+#include "sdk_p.h"
+
+#include <QDir>
+#include <QRegularExpression>
+
+namespace  {
+    const char DOCKER[] = "docker";
+    const char IMAGE[] = "image";
+    const char LIST[] = "ls";
+    const char FORMAT[] = "--format";
+    const char IMAGE_FORMAT[] = "{{.Repository}}";
+    const char UNNAMED_IMAGE[] = "<none>";
+}
+
+namespace Sfdk {
+
+DockerManager::DockerManager(QObject *parent)
+    : QObject(parent)
+{
+
+}
+
+DockerManager *DockerManager::s_instance = nullptr;
+
+DockerManager *DockerManager::instance()
+{
+    if (!s_instance) {
+        s_instance = new DockerManager();
+    }
+    return s_instance;
+}
+
+QStringList DockerManager::listedImages(const QString &output)
+{
+    QStringList images;
+    QStringList lines = output.split(QRegularExpression("[\r\n]"), QString::SkipEmptyParts);
+    for (QString line : lines) {
+        if (line != QLatin1String(UNNAMED_IMAGE))
+            images.append(line);
+    }
+    return images;
+}
+
+void DockerManager::fetchRegisteredVirtualMachines(const QObject *context, const Functor<const QStringList &, bool> &functor)
+{
+    Q_ASSERT(context);
+    Q_ASSERT(functor);
+
+    QStringList arguments;
+    arguments.append(QLatin1String(IMAGE));
```

Comment:
Some more cases for QLatin1String elimination?

Reply:
fixed